### ISSUE TITLE
Type annotations cause RuntimeException

### ISSUE THREAD

**Issue by tomvandenberge** [schauder/degraph#55]
Hi Jens,

I'm trying to get your tool running, because it looks very useful to me. However, a very simple test like 

`assertThat(classpath(), is(violationFree()));`

results in the following exception

```
java.lang.RuntimeException
	at org.objectweb.asm.MethodVisitor.visitTypeAnnotation(Unknown Source)
	at org.objectweb.asm.ClassReader.b(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at de.schauderhaft.degraph.analysis.asm.Analyzer$.de$schauderhaft$degraph$analysis$asm$Analyzer$$analyze$1(Analyzer.scala:18)
	...
```

It seems like asm has problems handing Java 8 type annotations that I'm using in my code.

**Comment by schauder** [comment]
Hi Tom, thanks for letting me know about this problem. 

I haven't tested Degraph for Java 8. I'll look into. Of course if you find out more about the reason and possible fixes, I'd appreciate an update (or a pull request

**Comment by tomvandenberge** [comment]
I think it has to do with the API version that is used to create the MethodVisitor (or FieldVisitor). API version 4 does not support Java 8 type annotations, but version 5 does. It seems that you are creating a ClassVisitor with version 4, in GraphBuildingClassVisitor:

`class GraphBuildingClassVisitor(g: Graph) extends ClassVisitor(Opcodes.ASM4)`

I haven't tried it, but changing it to `Opcodes.ASM5` might fix it.

**Comment by tomvandenberge** [comment]
I don't have a problem making and testing the fix myself, but I'm having problems building the project. When running `gradlew installApp`, it fails with

`No such property: ossrhUsername for class: org.gradle.api.publication.maven.internal.ant.DefaultGroovyMavenDeployer`

I don't have any gradle or groovy knowledge, so hopefully you can tell me how to fix this ;)

**Comment by schauder** [comment]
Give me a second to reproduce THAT problem

**Comment by schauder** [comment]
Hmm, I can't reproduce the problem locally.

Can you please try the following:

1. run 'gradlew test' this should compile everything and run the tests

2. create a file gradle.properties in the .gradle directory under your home directory and put the following content in:

    signing.keyId=xxx
    signing.password=xxx
    signing.secretKeyRingFile=xxx

    ossrhUsername=xxx
    ossrhPassword=xxx

    bintrayApikey=xxx

try again to run the test and installApp

As you might have guessed these properties are need to deploying degraph to maven central and other repositories. You shouldn't really need them and I just now ran the build with those properties removed, so to be honest I don't know what is going on there.

**Comment by tomvandenberge** [comment]
Adding the gradle.properties file makes it possible to run `gradlew test` and `gradlew installApp`. However, `installApp` does not seem to install the artifacts in the maven repository. In order to test my fix, I need to be able to install the maven artifacts in my local maven repository. I've tried `gradlew install`, but this fails with the error message 

`Unable to retrieve secret key from key ring file '/Users/tom/git/degraph/check/xxx' as it does not exist`

I don't understand why a secret key is required to install to the local repository. Any ideas?

**Comment by schauder** [comment]
Oh dear, I guess that kind of stuff happens when a single developer works on a project.

The secret key is really just for uploading to maven central and shouldn't be needed for a simple install.

One workaround would be: run installApp. In the directory degraph\build\install\degraph\lib you'll find all the jars, and you could manually replace them in your local repository.

I also try to figure out why it is trying to access all these properties.

**Comment by schauder** [comment]
I created and pushed a branch, that does not try to sign anything. Give it a try if you want.

**Comment by tomvandenberge** [comment]
Using this new branch doesn't change anything; I'm still getting the same error messages.

I'm also not able to find the jar file after running `installApp`. The directory `degraph/build/install` does not exist. There is a `degraph/build/libs/degraph-0.1.2.jar`, but it does not get updated when I run `installApp` after making changes to `GraphBuildingClassVisitor.scala`.

**Comment by schauder** [comment]
Hmm, doesn't look like I can do anything quick about the build problems. I'll look into that later. For now  I'll will take care of the Java 8 issue myself. Unfortunately this will take a couple of days.

Sorry for the inconvinience

**Comment by tomvandenberge** [comment]
No worries, take your time.

**Comment by schauder** [comment]
Can you try once more to run test/installApp, based on the current master? Please make sure to remove all properties from your gradle properties file. It should build now.

**Comment by tomvandenberge** [comment]
Great, it works! 
Thanks Jens. I'll fork your project, make the changes to support Java 8, and make a pull request.

**Comment by schauder** [comment]
Very much appreciated.

**Comment by schauder** [comment]
fixed by #58

Note that this fixes only the RuntimeException. It (probably) does not analyze Type Annotations correctly
